pkg/volume/cinder/cinder.go

```diff
@@ -143,13 +145,38 @@ func (plugin *cinderPlugin) NewProvisioner(options volume.VolumeOptions) (volume
 	return plugin.newProvisionerInternal(options, &CinderDiskUtil{})
 }
 
+func (plugin *cinderPlugin) parseProvisioningOptions(options volume.VolumeOptions) (cinderProvisioningOptions, error) {
+	var opts cinderProvisioningOptions
+
+	optStr := strings.TrimSpace(options.ProvisioningOptions)
+	if optStr != "" {
+		bytes := []byte(optStr)
+
+		err := json.Unmarshal(bytes, &opts)
+		if err != nil {
+			return opts, fmt.Errorf("error parsing configuration of %s provisioning plugin: %v", plugin.Name(), err)
```

Comment:
I don't know, it can be pretty long JSON string. I am not fan of code in error messages.